src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/CompletionSource.cs

```diff
@@ -69,7 +69,15 @@ internal CompletionSource(ITextView textView, IThreadingContext threadingContext
                 return AsyncCompletionData.CompletionStartData.DoesNotParticipateInCompletion;
             }
 
-            if (!document.Project.Solution.Workspace.Options.GetOption(CompletionOptions.BlockForCompletionItems, service.Language))
+            var workspace = document.Project.Solution.Workspace;
+
+            // LiveShare does not want Roslyn to participate in completion if the workspace is of RemoteLanguageServiceWorkspace.
```

Comment:
why?

Reply:
1.  That would be a good explanation to put in the comments.
2. It's unclear why this had to be added now.  Didn't LiveShare get added a while back?
3. This seems hinky to me.  Having Roslyn need to know about this seems busted.  Imagine if 'LiveShare' was a 3rd party extension.   Having Roslyn hardcode this in would be a very leaky abstraction.

Can we not jsut fix this by having completion import the appropriate completion providers, and then having this workspace just export an empty set of providers (or something similar to that)?